docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,363 @@
+    // valid. If the state transition is invalid, a non-nil error should be
+    // returned.
+    //
+    // It is guaranteed that the Parent has been successfully verified.
+    Verify() error
+
+    // Bytes returns the binary representation of this block.
+    //
+    // This is used for sending blocks to peers. The bytes should be able to be
+    // parsed into the same block on another node.
+    Bytes() []byte
+
+    // Height returns the height of this block in the chain.
+    Height() uint64
+}
+```
+
+### `choices.Decidable`
+
+This interface is the superset of every decidable object, such as transactions, blocks, and vertices. You can see the full file from [here.](https://github.com/ava-labs/avalanchego/blob/v1.7.4/snow/choices/decidable.go)
+
+```go title="/snow/choices/decidable.go"
+// Decidable represents element that can be decided.
+//
+// Decidable objects are typically thought of as either transactions, blocks, or
+// vertices.
+type Decidable interface {
+	// ID returns a unique ID for this element.
+	//
+	// Typically, this is implemented by using a cryptographic hash of a
+	// binary representation of this element. An element should return the same
+	// IDs upon repeated calls.
+	ID() ids.ID
+
+	// Accept this element.
+	//
+	// This element will be accepted by every correct node in the network.
+	Accept() error
+
+	// Reject this element.
+	//
+	// This element will not be accepted by any correct node in the network.
+	Reject() error
+
+	// Status returns this element's current status.
+	//
+	// If Accept has been called on an element with this ID, Accepted should be
+	// returned. Similarly, if Reject has been called on an element with this
+	// ID, Rejected should be returned. If the contents of this element are
+	// unknown, then Unknown should be returned. Otherwise, Processing should be
+	// returned.
+	Status() Status
+}
+```
+
+## rpcchainvm
+
+`rpcchainvm` is a special VM that wraps a `block.ChainVM` and allows the wrapped blockchain to run in its own process separate from AvalancheGo. `rpcchainvm` has two important parts: a server and a client. The [server](https://github.com/ava-labs/avalanchego/blob/master/vms/rpcchainvm/vm_server.go) runs the underlying `block.ChainVM` in its own process and allows the underlying VM's methods to be called via gRPC. The [client](https://github.com/ava-labs/avalanchego/blob/master/vms/rpcchainvm/vm_client.go) runs as part of `AvalancheGo` and makes gRPC calls to the corresponding server in order to update or query the state of the blockchain.
+
+To make things more concrete: suppose that `AvalancheGo` wants to retrieve a block from a chain run in this fashion. `AvalancheGo` calls the client's `GetBlock` method, which makes a gRPC call to the server, which is running in a separate process. The server calls the underlying VM's `GetBlock` method and serves the response to the client, which in turn gives the response to `AvalancheGo`.
```

Comment:
inline with previous comments I don't think this section is necessary

Reply:
This might be a little information that is cool to share, on how AvalancheGo can interact with the VM irrespective of its programming language.

Reply:
I agree this is valuable to include. Can we break this down into a step by step instead of a paragraph form to make it easier to follow?